packages/xen-api/src/_parseUrl.js

```diff
@@ -1,12 +1,20 @@
-const URL_RE = /^(?:(https?:)\/*)?(?:([^:]+):([^@]+)@)?([^/]+?)(?::([0-9]+))?\/?$/
+const URL_RE = /^(?:(https?:)\/*)?(?:([^:]+):([^@]+)@)?(?:\[([^\]]+)\]|([^:/]+))(?::([0-9]+))?\/?$/
 
 export default url => {
   const matches = URL_RE.exec(url)
   if (matches === null) {
     throw new Error('invalid URL: ' + url)
   }
 
-  const [, protocol = 'https:', username, password, hostname, port] = matches
+  const [
+    ,
+    protocol = 'https:',
+    username,
+    password,
+    ipv6,
+    hostname = ipv6,
```

Comment:
Maybe you should add a comment explaining why `hostname` is `ipv6` by default.

Reply:
Hmm, the hostname is not IPv6 by default, it's just that there are two capture groups in the regexp for the hostname, one that matches IPv6 (which uses a different format) and the other that matches the other formats.

The default value here is used to combine both of these groups into a single var.

What do you think would be a good comment for this code?

Reply:
It's not trivial to explain and is, IMHO, easy to understand for someone who would maintain this code (and the regexp).

If you don't mind, let's leave it as it is, we'll see in the future if we were wrong about it :wink: